console-src/src/enhancers/withOAuthClients.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { Component } from "react";
+import { Client } from "@kontist/client";
+
+import prodConfig from "../config";
+import devConfig from "../config.dev";
+
+import Loading from "../pages/Loading";
+
+import {
+  OAuthClient,
+  CreateOAuthClientPayload,
+  UpdateOAuthClientPayload,
+  DeleteOAuthClientPayload
+} from "../types/oAuthClient";
+import {
+  fetchClientsQuery,
+  createClientMutation,
+  updateClientMutation,
+  deleteClientMutation
+} from "../queries";
+
+const { baseAPIUrl, clientId, redirectUri } =
+  process.env.NODE_ENV === "production" ? prodConfig : devConfig;
+
+const STATE_KEY = "state";
+const VERIFIER_KEY = "verifier";
+const CLIENT_SCOPE = "clients";
+const CODE_QUERY_PARAM = "code";
+
+sessionStorage.setItem(
+  STATE_KEY,
+  sessionStorage.getItem(STATE_KEY) || (Math.random() + "").substring(2)
+);
+sessionStorage.setItem(
+  VERIFIER_KEY,
+  sessionStorage.getItem(VERIFIER_KEY) || (Math.random() + "").substring(2)
+);
+
+const client = new Client({
+  baseUrl: baseAPIUrl,
+  clientId,
+  redirectUri,
+  scopes: [CLIENT_SCOPE],
+  state: sessionStorage.getItem(STATE_KEY) || "",
+  verifier: sessionStorage.getItem(VERIFIER_KEY) || ""
+});
+
+interface OAuthClientsContext {
+  isLoading: boolean;
+  oAuthClients: OAuthClient[];
+  createClient: (payload: CreateOAuthClientPayload) => Promise<Object> | void;
+  updateClient: (payload: UpdateOAuthClientPayload) => Promise<Object> | void;
+  deleteClient: (payload: DeleteOAuthClientPayload) => Promise<Object> | void;
+}
+
+interface State extends OAuthClientsContext {
+  isAuthenticated: boolean;
+}
+
+const { Provider, Consumer } = React.createContext<OAuthClientsContext>({
+  isLoading: false,
+  oAuthClients: [],
+  createClient: () => {},
+  updateClient: () => {},
+  deleteClient: () => {}
+});
+
+class OAuthClientsProvider extends Component<{}, State> {
```

Comment:
It looks like this component has two responsibilities:
1. Authorize user and fetch access token
2. Provide context to handle oauth clients

Can these responsibilities be separated into 2 components?

Reply:
Yeah it is certainly possible and it also looks more reasonable to me, I will change it 👍

Reply:
Yeah, this separation is even better. Good job! 👍